frontend/src/metabase/query_builder/components/NativeQueryEditor/CodeMirrorEditor/CodeMirrorEditor.tsx

```diff
@@ -50,6 +51,12 @@ export const CodeMirrorEditor = forwardRef<
 
   const extensions = useExtensions(query);
 
+  const engine = Lib.engine(query);
+  const engineType = getEngineNativeType(engine ?? undefined);
+  const SQLPlaceholder = "SELECT * FROM TABLE_NAME";
+  const JSONPlaceholder = `[ { "$project": { "_id": "$_id" } } ]`;
+  const placeholder = engineType === "sql" ? SQLPlaceholder : JSONPlaceholder;
+
```

Comment:
I would move all this to a helper function like `getPlaceholder(query)`.

Also, are there only two engine types? Maybe falling back to the empty string would be better. Come to think of it, I'm pretty sure MongoDB is not the only JSON engine type, so you might want to match on Mongo specifically because the placeholder here is very specific to Mongo.

Reply:
https://github.com/metabase/metabase/pull/54438/commits/35f628614fe2f6d5c235890f10932a8b57c4f4e8